common/src/test/java/com/google/auto/common/MoreTypesIsTypeOfTest.java

```diff
@@ -15,198 +15,171 @@
  */
 package com.google.auto.common;
 
-import static com.google.common.collect.Iterables.getOnlyElement;
 import static com.google.common.truth.Truth.assertWithMessage;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
-import com.google.common.collect.Iterables;
+import com.google.common.collect.ImmutableList;
 import com.google.testing.compile.CompilationRule;
-import javax.lang.model.element.Element;
+import java.util.List;
+import java.util.SortedMap;
 import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.TypeKind;
 import javax.lang.model.type.TypeMirror;
 import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
-/**
- * Tests {@link MoreTypes#isTypeOf}.
- */
+/** Tests {@link MoreTypes#isTypeOf(Class, TypeMirror)}. */
 @RunWith(JUnit4.class)
 public class MoreTypesIsTypeOfTest {
 
   @Rule public CompilationRule compilationRule = new CompilationRule();
 
-  private Elements elements;
+  private Elements eltUtils;
+  private Types typeUtils;
 
   @Before
   public void setUp() {
-    this.elements = compilationRule.getElements();
+    this.eltUtils = compilationRule.getElements();
+    this.typeUtils = compilationRule.getTypes();
   }
 
-  private interface TestType {}
+  @Test
+  public void isTypeOf_primitiveAndBoxedPrimitiveTypes() {
+    class PrimitiveTypeInfo {
+      final Class<?> CLASS_TYPE;
+      final Class<?> BOXED_CLASS_TYPE;
+      final TypeKind TYPE_KIND;
+
+      PrimitiveTypeInfo(Class<?> classType, Class<?> boxedClassType, TypeKind typeKind) {
+        this.CLASS_TYPE = classType;
+        this.BOXED_CLASS_TYPE = boxedClassType;
+        this.TYPE_KIND = typeKind;
+      }
+    }
+    final List<PrimitiveTypeInfo> primitivesTypeInfo =
+        ImmutableList.of(
+            new PrimitiveTypeInfo(Byte.TYPE, Byte.class, TypeKind.BYTE),
+            new PrimitiveTypeInfo(Short.TYPE, Short.class, TypeKind.SHORT),
+            new PrimitiveTypeInfo(Integer.TYPE, Integer.class, TypeKind.INT),
+            new PrimitiveTypeInfo(Long.TYPE, Long.class, TypeKind.LONG),
+            new PrimitiveTypeInfo(Float.TYPE, Float.class, TypeKind.FLOAT),
+            new PrimitiveTypeInfo(Double.TYPE, Double.class, TypeKind.DOUBLE),
+            new PrimitiveTypeInfo(Boolean.TYPE, Boolean.class, TypeKind.BOOLEAN),
+            new PrimitiveTypeInfo(Character.TYPE, Character.class, TypeKind.CHAR));
+
+    for (int ki = 0; ki < 2; ki++) { // ki = 0: primitives, ki = 1: boxed primitives
```

Comment:
I think it would be clearer to have something like:

```java
for (boolean boxedI : new boolean[] {false, true})
```
and likewise for `boxedJ` below.

Reply:
Beautiful. :)